### DISCUSSION TITLE
Switchable Graphics not working on osu!Lazer (Laptop)

### DISCUSSION THREAD

**Discussion by sortedcord** [ppy/osu#14876] [Q&A] [answered]
As a heads up, I wanted to explain why I want to change the default GPU in the first place. This is a issue mainly occurring in Budget-Mid Range HP Laptops. Most models have dedicated GPUs (Mainly Radeon GPUs) that underperform integrated GPUs ( >= Skylake CPUs). The same was the case with my laptop. They have put a R5 M330 along with a Intel HD 620. 

For comparison, while using dedicated GPU, at max I get 125-175 fps (no vsync) along with stutters, fps drops and high battery usage. On the other hand, while using intel hd 620, I get anywhere from 300-400 fps which is more or less stable without any stutters. Overall a better experience (these comparisons were made from the stable release).

The issue is that I cannot seem to change the GPU used in windows. Here is a screenshot with no set graphic rules:-

![image](https://user-images.githubusercontent.com/37407370/135036357-79d2afba-5c08-49ed-82b9-ca7b7faa51a7.png)

And in the processes tab, it clearly shows that osu! is using `GPU1` which is the dedicated GPU, which is expected because windows automatically uses the dedicated GPU for any application which has a gaming profile...

![image](https://user-images.githubusercontent.com/37407370/135036527-5a63e0d7-e46c-4289-bebc-8254633a4f24.png)

Then, I went ahead to use the integrated graphics-

![image](https://user-images.githubusercontent.com/37407370/135036979-21738ae3-81a3-4fd0-8c88-ee9b6083e120.png)

and opened lazer again, but it was still using `GPU1` instead of `GPU0` (Intel HD). I did a restart still it uses the `GPU1` instead of `GPU0`....

![image](https://user-images.githubusercontent.com/37407370/135037178-d77ffcc5-2f94-4940-91b1-74d2589c343f.png)

I am certain this is the way to switch graphics, but im not sure if this is an issue with lazer or windows because I was able to do the same with stable and it worked without issues...

**Comment by smoogipoo** [comment]
I can only think that this is an SDL2 issue. We're not currently forcing the dedicated GPU at our end.

**Reply by sortedcord** [reply]
So, there isn't a way to switch graphics?

**Reply by smoogipoo** [reply]
If it doesn't work through windows, then no.

**Comment by sortedcord** [comment, marked as answer]
With a bit of digging around in the installation path, I was able to switch graphics again. Turns out that you are supposed to apply the display rules to the executable present in the folder of the version. Something like `C:\User\Appdata\Local\osulazer\app-2021.xxx.x\osu.exe` instead of `C:\User\Appdata\Local\osulazer\osu.exe`. I applied the settings to the one in the version folder, and it worked.

![image](https://user-images.githubusercontent.com/37407370/135249775-999422a6-3beb-4a83-8166-048df6c7506c.png)

![image](https://user-images.githubusercontent.com/37407370/135249943-9fbaa35a-2a04-4029-89ba-6f97ff93ffb6.png)

**Reply by gagahpangeran** [reply]
Related https://github.com/ppy/osu/issues/10550
